src/app-layer.c

```diff
@@ -460,6 +460,11 @@ static int TCPProtoDetect(ThreadVars *tv, TcpReassemblyThreadCtx *ra_ctx,
                 SCLogDebug("reversing flow after proto detect told us so");
                 PacketSwap(p);
                 FlowSwap(f);
+                // Reset signature groups
```

Comment:
This looks incomplete, see https://github.com/OISF/suricata/blob/master/src/detect.c#L930 for what is needed when we switch detect engine in a flow.

Reply:
perhaps it would be enough to set `Flow::de_ctx_version` to `UINT32_MAX` or something.

Reply:
Oh, do you mean like `pflow->flowvar = NULL;` ?

Will do